fatal error: Eigen/Core: No such file or directory

Hello,

I've:

* Read the documentation to build with the latest source
* No other person has already created the same issue with current build
* Using cut-n-paste below to the extent practical
* Reporting a single issue

##### System information (version)
matha@myUbuntu:~/projects/opencv/build$ uname -a
Linux BeUlta 5.0.0-17-generic #18-Ubuntu SMP Tue Jun 4 15:34:08 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux

##### Detailed description

```
[ 14%] Building CXX object modules/core/CMakeFiles/opencv_test_core_pch_dephelp.dir/opencv_test_core_pch_dephelp.cxx.o
In file included from /home/chowkidar/projects/opencv/modules/core/test/test_precomp.hpp:12,
                 from /home/chowkidar/projects/opencv/build/modules/core/opencv_test_core_pch_dephelp.cxx:1:
/home/chowkidar/projects/opencv/modules/core/include/opencv2/core/private.hpp:66:12: fatal error: Eigen/Core: No such file or directory
 #  include <Eigen/Core>
            ^~~~~~~~~~~~
compilation terminated.
make[2]: *** [modules/core/CMakeFiles/opencv_test_core_pch_dephelp.dir/build.make:63: modules/core/CMakeFiles/opencv_test_core_pch_dephelp.dir/opencv_test_core_pch_dephelp.cxx.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:2715: modules/core/CMakeFiles/opencv_test_core_pch_dephelp.dir/all] Error 2
make: *** [Makefile:163: all] Error 2
chowkidar@BeUlta:~/projects/opencv/build$ 
```

How to get past this error for the make command? (all steps performed documented in attachment).

Thanks.
[OpenCV installation.txt](https://github.com/opencv/opencv/files/3317015/OpenCV.installation.txt)

P.S. I have built OpenCV on many platforms previously.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -475,7 +475,7 @@ OCV_OPTION(INSTALL_TESTS            "Install accuracy and performance test binar
 # OpenCV build options
 # ===================================================
 OCV_OPTION(ENABLE_CCACHE              "Use ccache"                                               (UNIX AND NOT IOS AND (CMAKE_GENERATOR MATCHES "Makefile" OR CMAKE_GENERATOR MATCHES "Ninja")) )
-OCV_OPTION(ENABLE_PRECOMPILED_HEADERS "Use precompiled headers"                                  ON IF (MSVC OR (NOT IOS AND NOT CMAKE_CROSSCOMPILING) ) )
+OCV_OPTION(ENABLE_PRECOMPILED_HEADERS "Use precompiled headers"                                  MSVC IF (MSVC OR (NOT IOS AND NOT CMAKE_CROSSCOMPILING) ) )
 OCV_OPTION(ENABLE_SOLUTION_FOLDERS    "Solution folder in Visual Studio or in other IDEs"        (MSVC_IDE OR CMAKE_GENERATOR MATCHES Xcode) )
 OCV_OPTION(ENABLE_PROFILING           "Enable profiling in the GCC compiler (Add flags: -g -pg)" OFF  IF CV_GCC )
 OCV_OPTION(ENABLE_COVERAGE            "Enable coverage collection with  GCov"                    OFF  IF CV_GCC )

